django/template/base.py

```diff
@@ -86,7 +86,7 @@
 # Match BLOCK_TAG_*, VARIABLE_TAG_*, and COMMENT_TAG_* tags and capture the
 # entire tag, including start/end delimiters. Using re.compile() is faster
 # than instantiating SimpleLazyObject with _lazy_re_compile().
-tag_re = re.compile(r"({%.*?%}|{{.*?}}|{#.*?#})")
+tag_re = re.compile(r"({%(?=.*%}).*?%}|{{(?=.*}}).*?}}|{#(?=.*#}).*?#})")
```

Comment:
Could we convert to verbose mode (`re.VERBOSE`) while we do this? This line is now punctuation soup... :)

Reply:
> punctuation soup

So, a regex? :wink:

Reply:
yep good thought! I'll address suggestions soon, sorry this fell off the radar for a bit